src/commands/NewIntegrationFileCommand.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as kamel from './../kamel';
+
+const validFilename = require('valid-filename');
+
+const LANGUAGES = ['Java', 'xml', 'Yaml', 'Groovy', 'JavaScript', 'Kotlin'];
+
+export async function create() {
+	let language = await vscode.window.showQuickPick(LANGUAGES);
+	if (language) {
+		let workspaceFolder = await vscode.window.showWorkspaceFolderPick();
+		if (workspaceFolder) {
+			let filename = await vscode.window.showInputBox({
+				prompt: 'Please provide a name for the new file (without extension)',
+				validateInput: (name: string) => {
+					return validateFileName(name, language as string, workspaceFolder as vscode.WorkspaceFolder);
+				}
+			});
+			if (filename) {
+				let kamelExe = kamel.create();
+				let fp: string = computeFullpath(language, workspaceFolder, filename);
+				kamelExe.invoke(`init ${fp}`);
+			}
+		}
+	}
+}
+
+function computeFullpath(language: string, workspaceFolder: vscode.WorkspaceFolder, filename: string) {
+	let extension = getFileExtensionForLanguage(language);
+	return path.join(workspaceFolder.uri.fsPath, `${filename}.${extension}`);
```

Comment:
path works fine with string folders too. Maybe its worth to change the parameter to string instead of WorksapceFolder? Where do u see the advantage in being that strict?

Reply:
We are retrieving a WorkspaceFolder from `vscode.window.showWorkspaceFolderPick()`
it factorizes the `workspaceFolder.uri.fsPath`, otherwise we would have to call it before calling the `computeFullpath` method.